src/cxx_supportlib/ServerKit/HttpHeaderParser.h

```diff
@@ -118,20 +119,35 @@ class HttpHeaderParser {
 		}
 	}
 
-	static size_t http_parser_execute_and_handle_pause(http_parser *parser,
-		const http_parser_settings *settings, const char *data, size_t len,
-		bool &paused)
+	static size_t http_parser_execute_and_handle_pause(llhttp_t *parser,
+		const char *data, size_t len, bool &paused)
 	{
-		size_t ret = http_parser_execute(parser, settings, data, len);
-		paused = len > 0 && ret != len && HTTP_PARSER_ERRNO(parser) == HPE_PAUSED;
-		if (paused) {
-			http_parser_pause(parser, 0);
-			http_parser_execute(parser, settings, data + len - 1, 1);
+		llhttp_errno_t rc = llhttp_get_errno(parser);
+		switch (rc) {
+		case HPE_PAUSED_UPGRADE:
+			llhttp_resume_after_upgrade(parser);
+			goto happy_path;
+		case HPE_PAUSED:
+			llhttp_resume(parser);
+			goto happy_path;
+		case HPE_OK:
+		happy_path:
+			switch (llhttp_execute(parser, data, len)) {
+			case HPE_PAUSED_H2_UPGRADE:
+			case HPE_PAUSED_UPGRADE:
+			case HPE_PAUSED:
+				paused = true;
+				return (llhttp_get_error_pos(parser) - data);
+			case HPE_OK:
+				return len;
+			}
+			// no default, fallthrough to error handling instead
```

Comment:
I get warnings here.

>  warning: 32 enumeration values not handled in switch: 'HPE_INTERNAL', 'HPE_STRICT', 'HPE_LF_EXPECTED'... [-Wswitch]

Suggested fix:

```diff
                                return (llhttp_get_error_pos(parser) - data);
                        case HPE_OK:
                                return len;
+                       default:
+                               return (llhttp_get_error_pos(parser) - data);
                        }
-                       // no default, fallthrough to error handling instead
         default:
                        return (llhttp_get_error_pos(parser) - data);
                }
```

Reply:
Yes, I know I'm not covering all cases, that's deliberate, thus the comment. The warning is not correct in this case.

Reply:
Wouldn't it be cleaner to have a default there as suggested, instead of using pragmas?